src/core/server/elasticsearch/client/cluster_client.ts

```diff
@@ -70,18 +71,23 @@ export class ClusterClient implements ICustomClusterClient {
   constructor(
     private readonly config: ElasticsearchClientConfig,
     logger: Logger,
+    private readonly auditorFactory: AuditorFactory,
     private readonly getAuthHeaders: GetAuthHeaders = noop
   ) {
     this.asInternalUser = configureClient(config, { logger });
     this.rootScopedClient = configureClient(config, { logger, scoped: true });
   }
 
   asScoped(request: ScopeableRequest) {
-    const scopedHeaders = this.getScopedHeaders(request);
+    const id = `scoped-client-${counter++}`;
     const scopedClient = this.rootScopedClient.child({
-      headers: scopedHeaders,
+      headers: this.getScopedHeaders(request),
+      name: id,
     });
-    return new ScopedClusterClient(this.asInternalUser, scopedClient);
+    const internalClient = this.asInternalUser.child({ name: id });
```

Comment:
btw `Symbol()` would address this problem as well. Unfortunately, the `child` interface doesn't support them.

Reply:
I'm not sure I follow, what do you mean?

Reply:
I have to generate uniq Id for every child. For that I have to use counter. I could get rid of `counter` if `name` accepted `symbol` 
```js
const id = Symbol();
const internalClient = this.asInternalUser.child({ name: id });
internalClient.on('request', (err: ApiError, event: RequestEvent) => {
        if (event.meta.name === id) {...
```

Reply:
Oks! This can be easily added :)